test/helper/common.bash

```diff
@@ -181,18 +181,15 @@ function wait_until_change_detection_event_completes() {
     [[ $(__change_event_status) == "${CHANGE_EVENT_END}" ]]
   }
 
-  if [[ ! $(__is_changedetector_processing) ]]
+  # A new change event is expected,
+  # If the last event status is not yet `CHANGE_EVENT_START`, wait until it is:
+  if [[ $(__is_changedetector_processing) -ne 0 ]]
```

Comment:
~~Does the omission of the `[[ ]]` make a difference here?~~ (**EDIT:** Nope). Otherwise I guess it'll produce the same problem.

When I was debugging, the first call to `__is_changedetector_processing()` was `0` (I echo appended the result to a file), and then inside the if condition, the 60 seconds of waiting on it to become true (again?) never happened, as it was always `1` (completed). Presumably because the `reload` was too fast? :sweat_smile: 

I suppose I could add a `sleep 1` prior to the conditional?

---

**UPDATE:** Results in same reproducible failure:

- Conditional check `if ! $(__is_changedetector_processing)` is getting treated as true
- All calls after that remain false :man_shrugging: 

This is specifically with `make test/setup-cli` and the `checking setup.sh: setup.sh email add and login` and `checking setup.sh: setup.sh email update` test cases, it's not failing 100% of the time, so likely some race condition.

Reply:
Identified as a bug with order of checks in the `email add and login` test.

Reply:
Not quite :sweat_smile: 

Change detection is now happening in 0-1 seconds vs the 5+ seconds incurred from restarts (_which is technically a bit longer in some cases until the services would be ready again IIRC_). Hence the race condition :|

It is passing more reliably now, but this could cause some false positive failures until the detection handling is improved for that speed :sweat_smile: (_finished before the 2nd event status check (in the conditional) has a chance to catch the `Change detected` log between polled checks_).